wherescape/connectors/gitlab/gitlab_wrapper.py

```diff
@@ -0,0 +1,198 @@
+
+        Returns:
+        response object: response of the request made
+        """
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self.access_token}",
+        }
+        response = requests.request(method, url, data=payload, headers=headers)
+        return response
+
+    def format_url(self, resource_api, page_variables, simple, order_by, since):
+        """Format URL
+
+        Parameters:
+        resource_api (string): The location of the resource requested
+        page_variables (object): {
+            "per_page": string or int,
+            "current_page": string or int
+        }
+        simple (boolean): If the response of Gitlab should be simplified this needs to be set on True
+        since (string): ISO formatted datetime string to indicate since which date you want values back
+
+        Returns:
+        Formatted url which can be used to make the request
+        """
+        updated_since = f"&updated_after={since}" if since else ""
+        pagination = f"per_page={page_variables['per_page']}&page={int(page_variables['current_page'])+1}"
+        return f"{self.base_url}/{resource_api}?order_by={order_by}&sort=asc&simple={simple}&{pagination}{updated_since}"
+
+    def paginate_through_resource(
+        self,
+        resource_api,
+        keys_to_keep,
+        since=None,
+        per_page=50,
+        simple="false",
+        order_by="id",
+    ):
+        """Paginate through resources
+        Since the Gitlab API has pagination, this helper function will paginate through the resource API.
+        It will do that until all responses are collected.
+        It cleans the response immediately and turns it into a tuple.
+        It does expect a list with objects in the response of the API.
+
+        Parameters:
+        resource_api (string): The location of the resource requested
+        keys_to_keep (list): List of keys returned by the API you want to keep
+        per_page (int): How many results per page you would like to get
+        simple (boolean): If the response of Gitlab should be simplified this needs to be set on True
+        since (string): ISO formatted datetime string to indicate since which date you want values back
+
+        Returns:
+        List of tuples with the values from the request
+
+        """
+        total_pages = 1
+        current_page = 0
+
+        all_resources = []
+
+        while current_page < total_pages:
+            page_variables = {"per_page": per_page, "current_page": current_page}
+            url = self.format_url(resource_api, page_variables, simple, order_by, since)
+
+            response = self.make_request(url, "GET")
+
+            if response.status_code == 403:
+                logging.info(
+                    f"{url}\n Forbidden resource. If you need this resource, please check the user's rights"
```

Comment:
The info statement seems not to print the URL : 
![image](https://user-images.githubusercontent.com/24241551/177713031-64aa3625-a456-432c-be30-4ae757140e10.png)